### PR TITLE
 SystemServer: Make system-mode=text the default in case of read failure

### PR DESCRIPTION
**Pull request by supercomputer7**
I know we should not encounter such case, but when I worked on #15593, I managed to remove the `/proc/system_mode` but replaced it with nothing so `SystemServer` launched `WindowServer` which crashed because it didn't know what to do without reading `system_mode`.

### CODE REVIEW

File: Userland/Services/SystemServer/main.cpp [hunk 1]
```diff
@@ -490,6 +494,8 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         TRY(determine_system_mode());
     }
 
+    dbgln("{}", g_system_mode);
```
Comment (bgianfo) [inline Userland/Services/SystemServer/main.cpp:497]: Isn't the double logging unnecessary, especially this line where we log no other context?

File: Userland/Services/SystemServer/main.cpp [hunk 2]
```diff
@@ -64,6 +64,9 @@ static void sigchld_handler(int)
 
 static ErrorOr<void> determine_system_mode()
 {
+    ArmedScopeGuard declare_text_mode_on_failure([&] {
+        g_system_mode = "text";
```
Comment (bgianfo) [inline Userland/Services/SystemServer/main.cpp:68]: The comments on line 74 and 79 are now wrong.

**Review by bgianfo — CHANGES_REQUESTED** [review]
Small nits

**Review by bgianfo — APPROVED** [review]

**Review by linusg — CHANGES_REQUESTED** [review]
This is consistently failing CI even after a restart, seems suspicious.

**Comment by supercomputer7** [comment]
I had a small mistake in the code (I removed the line where it was setting the system mode based on what we read from `/proc/system_mode`). Should be good to go now.

**Pull request merged**